### PR TITLE
Divorce biome lib and cleanup code

### PR DESCRIPTION
**Pull request by wsor4035**
see the title

* [ ] get rid of biome lib
* [x] fix .luacheckrc
* [x] add github action for luacheck
* [ ] replace global settings with settings api
* [ ] replace default_settings.txt file with descriptions in settings api
* [x] fix various luacheck warnings
* [ ] profit?

fixes https://github.com/mt-mods/moretrees/issues/9

### CODE REVIEW

**Comment by wsor4035** [comment]
note: this will need backwards compat testing to make sure nothing crashes on old worlds with this PR

**Comment by wsor4035** [comment]
tree generation does now work, note that it is laggy as all get out due to way to many trees being placed and spawning, uncomment the set node stone and that will keep them from spawning to play with placement values

**Comment by PrairieAstronomer** [comment]
Might be nice to go away with biomelib. I might try to port this to work with mineclone 2, but I dont know if I will for sure.

**Comment by wsor4035** [comment]
> Might be nice to go away with biomelib

more like required, given how terrible its perf is

> I might try to port this to work with mineclone 2

make a pr, be glad to make this game agnostic. 

above said, i do need to finish this....

**Review by BuckarooBanzay — APPROVED** [review]

**Comment by BuckarooBanzay** [comment]
LGTM :+1:  (ingame is not better or worse than before :wink:)

File: cocos_palm.lua [hunk 1]
```diff
@@ -82,8 +82,8 @@ minetest.register_abm({
 	nodenames = { "moretrees:palm_fruit_trunk_gen" },
 	interval = 6,
 	chance = 1,
-	min_y = biome_lib.mapgen_elevation_limit.min,
-	max_y = biome_lib.mapgen_elevation_limit.max,
+	min_y = -16,
+	max_y = 48,
```
Comment (Niklp09) [inline cocos_palm.lua:86]: +1 to +32 should be enough

File: init.lua [hunk 2]
```diff
@@ -95,111 +95,139 @@ dofile(modpath.."/saplings.lua")
...
+	end
+
+    if #locations == 0 then return end
+    for _, pos in ipairs(locations) do
+		--minetest.chat_send_all("yay")
+        local timer = minetest.get_node_timer({x=pos.x, y=pos.y+1, z=pos.z})
+        timer:start(math.random(2,10))
+		--minetest.set_node(pos, {name="default:stone"})
+    end
+end)
```
Comment (Niklp09) [inline init.lua:225]: `math.random` should be localised in on_generated callbacks

**Review by Niklp09 — COMMENTED** [review]
great work, wsor! I've added some comments where you can make optimizations

File: mod.conf [hunk 3]
```diff
@@ -1,4 +1,4 @@
 name = moretrees
-depends = default, biome_lib, vessels
```
Comment (Niklp09) [inline mod.conf:2]: ethereal should be an optional depend to keep trees registered (https://notabug.org/TenPlus1/ethereal/src/master/biomes_init.lua#L26)

File: cocos_palm.lua (hunk 1 above)
Comment (wsor4035) [reply cocos_palm.lua:86]: values taken straight from biome lib. this is invalid/scope creep. file and issue if you would like it changed

File: init.lua (hunk 2 above)
Comment (wsor4035) [reply init.lua:225]: doesnt need to be.

File: mod.conf (hunk 3 above)
Comment (wsor4035) [reply mod.conf:2]: looks like i need to have a word with tenplus1 again....

Comment (wsor4035) [reply mod.conf:2]: https://github.com/mt-mods/moretrees/pull/11/commits/8570aab0c91ca74ba8d443b4b345b0ea21475d69 resolved

**Review by Niklp09 — APPROVED** [review]
tested a bit ingame

File: init.lua (hunk 2 above)
Comment (wsor4035) [reply init.lua:225]: further more, math.random is used elsewhere across this mod. if you wish to optimize this, should be done in another pr that hits all instances as once

**Pull request merged**